### PR TITLE
First draft with some basic structure and content definition.

### PR DESCRIPTION
**Pull request by tribeiro**

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -1,57 +1,146 @@
...
 
-  and
+Introduction
+============
 
-  Subsubsections
-  ^^^^^^^^^^^^^^
+The main goal of the observatory control system is to provide a user-friendly framework with a high level of flexibility.
+During commissioning, engineering and operations there will be situations where users will want to execute non-standard, customized operations.
+The goal of the observatory control system is not only to allow users to perform those custom tasks but also to be able to incorporate them to the system.
```
Comment (patrickingraham) [inline index.rst]: to the system if desired to do so.  or if deem appropriate.. or if they will be run regularly.... you get the point.

File: index.rst [hunk 2]
```diff
@@ -1,57 +1,146 @@
...
+   :name: fig-scriptqueueui
+   :target: ../_images/ScriptQueueUI.png
+   :alt: LOVE ScriptQueue user interface
+
+   ScriptQueue LOVE user interface. ...
+
+In general, scripts require minimum interaction to be executed except, of course, for the occasional configuration.
+A database of configuration will be available for users to execute the most common set of operations and they also have the possibility of editing configuration on the user interface, which also provides schema validation.
+
+Regular operations are separated into two distinct groups of `SAL Scripts`_; standard and external.
```
Comment (patrickingraham) [inline index.rst]: I think "regular operations" really only refers to standardscripts, anything in external is not going to be a bit of a special case, no?

File: index.rst [hunk 3]
```diff
@@ -1,57 +1,146 @@
...
+.. _Contributing:
+
+Contributing
+============
+
+Occasionally, specially during early commissioning and integration activities, users may face situations where they need to perform a certain type of operation that is not possible with the available script set.
+In these situations, users are highly encouraged to contribute to the main feature set.
+
+.. note::
+   I feel that this is already covered in tstn-010, so, maybe remove it?
```
Comment (patrickingraham) [inline index.rst]: I think I'd just put a single sentence saying something like, "In cases where code developed in a notebook should be incorporated into the observatory software, the workflow in tstn-010 must be followed"

**Review by patrickingraham — CHANGES_REQUESTED** [review]
First off, thanks for doing this, it's a great start to the conversation.
So, thinking out loud here, I think it makes sense to break down this document by use-cases, where those use-cases are user-level dependent. I think a top-level description of LOVE, scriptQueue and Nublado are fine as a pre-amble.
I think the reason it is important to define users and their expected uses is because it's a middle ground between what is in tstn-010 and what will go into the scripting documentation (how-to's etc.)

Do we expect "standard operators" to run notebooks? Write notebooks? Will they only ever run things out of standardscripts? What about responding to errors, will they need to know how to restart CSCs? Change configs? kill and re-open containers?

For "standard commissioning users", they will write notebooks, and launch scripts, but will they write scripts? Or, will they only make small modifications to current scripts such that no futzing with dependencies in the scriptQueue are required? What level of troubleshooting will they perform? 

For "advanced users", I think the interesting line to draw is when you need to call for help. For example, if you want to write a new (or modify) a script on the fly then run it on the scriptqueue, they need to know how to do that. Where does this user's permissions and capabilities end?  

Probably worth having a slack call.... but thought I'd brain dump this here :)

File: index.rst [hunk 4]
```diff
@@ -1,57 +1,332 @@
...
-  -----------
+.. _Introduction:
 
-  and
+Introduction
+============
 
-  Subsubsections
-  ^^^^^^^^^^^^^^
+The construction, integration, commissioning and operation of observatories requires a significant amount of knowledge spanning many disciplines.
```
Comment (tribeiro) [inline index.rst]: > knowledge spanning many disciplines.

how about:

> knowledge spanning multiple disciplines.

File: index.rst [hunk 5]
```diff
@@ -1,57 +1,332 @@
...
+However, all operational aspects are performed using some piece of control software, whether it be embedded low-level servo control to enable precise positioning of the mount to a simple graphical display of the current weather conditions.
```
Comment (tribeiro) [inline index.rst]: This phrase is a bit hard to read. Maybe rephrase it? 

How about:

>  However, all operational aspects involves some level of control software; from embedded low-level servo control for precise positioning of the mount, to simple graphical display of the current weather conditions.

Maybe start a new paragraph after this?

File: index.rst [hunk 6]
```diff
@@ -1,57 +1,332 @@
...
+The observatory control system, which encompasses numerous levels of control, must span a large range of functionalities that are applicable to numerous users, all with varying experience levels and expectations regarding ease-of-use.
```
Comment (tribeiro) [inline index.rst]: As before, this phrase is also a bit hard to read. Maybe break it down into multiple phrases or rewrite it?

File: index.rst [hunk 7]
```diff
@@ -1,57 +1,332 @@
...
+Therefore, from within this centralized stack of code, a high degree of flexibility must be maintained, while providing a user-friendly framework to all levels of users.
```
Comment (tribeiro) [inline index.rst]: I think "centralized stack of code" may be too technical. I am not sure people not used to software development would know what a software stack is. 

Maybe something on these lines?

> Therefore, it is important for this software to maintain a high degree of flexibility, while providing a user-friendly framework for users at all levels.

File: index.rst [hunk 8]
```diff
@@ -1,57 +1,332 @@
...
-  To add images, add the image file (png, svg or jpeg preferred) to the
-  _static/ directory. The reST syntax for adding the image is
+  - Regular operations routines should be easily accessible;
+  - Must include a highly customizable scripting capability must be available and able to run user-written code;
+  - An interactive mode, equivalent to a user sandbox space, must be available to allow development of routines on the fly as well as debugging;
+  - Capability to incorporate user-developed features;
 
-  .. figure:: /_static/filename.ext
-     :name: fig-label
+In the standard operational model used to perform the survey, the observatory will be largely driven by the Scheduler, which will designate a visit script, which consists of a series of operations that are performed including slewing the telescope and taking an image.
```
Comment (tribeiro) [inline index.rst]: How about?

>In the standard operational model (e.g. survey mode), the observatory will be largely driven by the Scheduler. 
Specific SAL Scripts, which consists of a series of operations (such as slewing the telescope and taking images), will be added to the ScriptQueue by the Scheduler as needed.

File: index.rst [hunk 9]
```diff
@@ -1,57 +1,332 @@
...
+The script will be execution will be automated while the user will be able to observe (or interrupt) the progress via a graphical interface.
```
Comment (tribeiro) [inline index.rst]: There's a rogue "will be". 

Suggested editing:

> The SAL Script execution will be automated, and users will be able to monitor, pause or interrupt the progress via a graphical interface.

File: index.rst [hunk 10]
```diff
@@ -1,57 +1,332 @@
...
+It is expected that operators will be much more involved during the start-up and shutdown phases, as well as during afternoon calibrations.
+In the majority of cases, the operations will be routine and are expected to be done from a high-level of the observatory control software, as is discussed further in following sections.
```
Comment (tribeiro) [inline index.rst]: Suggested rephrasing:

> In most cases, these routine operations are expected to be executed from a high-level interface of the observatory control software (see discussion furthermore).

File: index.rst [hunk 11]
```diff
@@ -1,57 +1,332 @@
...
 
-     Caption text.
+During integration, commissioning, and engineering time, there will be situations where users will want to execute non-standard, customized operations which require an increased level of flexibility and control.
+As to be expected, the additional functionality required to perform such tasks requires more in-depth knowledge specific areas of the system and therefore require a knowledge base and toolset beyond what is typical of an observing specialist which performs nighttime operations.
```
Comment (tribeiro) [inline index.rst]: Needs some rephrasing... Suggestion:

> The additional functionality required to perform such tasks demand a more in-depth knowledge in specific areas of the system. 
Therefore, it is reasonable to expect that they require a knowledge base and toolset beyond what is typical of an observing specialist in charge of nighttime operations.

File: index.rst [hunk 12]
```diff
@@ -1,57 +1,332 @@
...
+This evolves further as one considers what tooling and knowledge is required for development of the system, however, all of the above users and usages require those interactive tools.
```
Comment (tribeiro) [inline index.rst]: Uhm.. not sure I understood what this means. I think the first part of the phrase is ok, and could be a single phrase on its own. Now, the second part I am not really sure.

Suggestion:

> This evolves further as one considers what tooling and knowledge is required for development of the system.
However, all of the above users and usages will, at some level, require these interactive tools.

File: index.rst [hunk 13]
```diff
@@ -1,57 +1,332 @@
...
 
-   Run: ``make html`` and ``open _build/html/index.html`` to preview your work.
-   See the README at https://github.com/lsst-sqre/lsst-technote-bootstrap or
-   this repo's README for more info.
+The breadth of the observatory control system allows users to perform both these routine and customized tasks via three main interactive components: the :ref:`LSST Observing Visualization Environment (LOVE)<LOVE>`, :ref:`ScriptQueue <Script-Queue>` and :ref:`Jupyter notebooks <Jupyter-notebooks>`.
```
Comment (tribeiro) [inline index.rst]: Suggested editing:

> The breadth of the observatory control system allows users to perform both standard routines and customized tasks via three main interactive components: the :ref:`LSST Observing Visualization Environment (LOVE)<LOVE>`, :ref:`ScriptQueue <Script-Queue>` and :ref:`Jupyter notebooks <Jupyter-notebooks>`.

File: index.rst [hunk 14]
```diff
@@ -1,57 +1,332 @@
...
+This document begins with a high-level overview of the systems and explore how users can take advantage of their features to accomplish different types of tasks
+References to more detailed documentation regarding their operation are linked throughout..
```
Comment (tribeiro) [inline index.rst]: The end-of-line period is duplicated.

File: index.rst [hunk 15]
```diff
@@ -1,57 +1,332 @@
...
 
-   Feel free to delete this instructional comment.
+Then, to assist in defining the expectations and knowledge levels associated with the multiple complexities associated with all of the required tasks that must be performed by the OCS, we've identified three main actors and their expected roles and ability levels associated their interactions with the control system.
```
Comment (tribeiro) [inline index.rst]: Suggested editing:

> Furthermore, we've identified three main actors, their expected roles and ability levels, associated with their interactions with the control system.
This should assist in defining the expectations and knowledge levels required to accomplish the tasks that must be performed through the control system.

File: index.rst [hunk 16]
```diff
@@ -1,57 +1,332 @@
...
+The identified actors are: 1) Observing Specialists, 2) Commissioning Scientists, 3) Software Developers.
```
Comment (tribeiro) [inline index.rst]: Maybe turn the list of actors into an action itemized list? I think this will help with readability.

File: index.rst [hunk 17]
```diff
@@ -1,57 +1,332 @@
...
+In reality, we expect (and need) several people to have significant overlap between these roles.
 
-:tocdepth: 1
+The documentation associated with the example procedures expected to be performed by each of the actors are then linked to provide a demonstration of the current usability level associated with each task.
 
-.. Please do not modify tocdepth; will be fixed when a new Sphinx theme is shipped.
+Under the standard operational procedures used to perform the survey, the observatory will be largely driven by the Scheduler, which will designate a visit script, which consists of a series of operations that are performed via a content controlled script. The script will be executed by the scriptQueue, and the user will be able to observe (or interrupt) the progress via the LOVE interface.
```
Comment (tribeiro) [inline index.rst]: Repeated paragraph. Remove it?

File: index.rst [hunk 18]
```diff
@@ -1,57 +1,332 @@
...
+It is expected that operators will be much more involved during the start-up and shutdown phases, as well as during afternoon calibrations.
+In the majority of cases, the operations will be routine and are expected to be done from a high-level of the observatory control software, as is discussed further in following sections.
 
-.. sectnum::
+Lastly, using concrete examples, we present examples of the types of interactions that these actors will perform with links to procedures on how to do them using the multiple interfaces.
```
Comment (tribeiro) [inline index.rst]: > Lastly, we present concrete examples of the types of interactions that these actors will perform, with links to procedures on how to execute them using the multiple interfaces.

File: index.rst [hunk 19]
```diff
@@ -1,57 +1,332 @@
...
+These examples explore how users can take advantage of the features also include cases of how *not* to use an interface due to its limitations or inefficiencies in some areas.
```
Comment (tribeiro) [inline index.rst]: Uhm... this phrase is a bit confusing. Can you rephrase it?

**Review by tribeiro — COMMENTED** [review]
Finished reviewing the first section. I'll continue to work on reviewing the next sections.

File: index.rst (hunk 6 above)
Comment (patrickingraham) [reply index.rst]: i just removed it.

File: index.rst (hunk 7 above)
Comment (patrickingraham) [reply index.rst]: Reworded as changing the above bits broke the flow.

File: index.rst (hunk 8 above)
Comment (patrickingraham) [reply index.rst]: I don't want to introduce those terms here... it's too early to be making references to specific things. 
I re-worded.

File: index.rst (hunk 5 above)
Comment (patrickingraham) [reply index.rst]: hrrm... don't really agree but I broke it up a little and started a new paragraph.

File: index.rst (hunk 10 above)
Comment (patrickingraham) [reply index.rst]: I didn't use this exactly but something like it.

File: index.rst (hunk 11 above)
Comment (patrickingraham) [reply index.rst]: Ok, so my sentence was definitely bad. I modified it

File: index.rst (hunk 12 above)
Comment (patrickingraham) [reply index.rst]: just removed most of this.

File: index.rst (hunk 17 above)
Comment (patrickingraham) [reply index.rst]: ooops, thanks.

File: index.rst (hunk 18 above)
Comment (patrickingraham) [reply index.rst]: rephrased.

File: index.rst (hunk 19 above)
Comment (patrickingraham) [reply index.rst]: rephrased

**Review by patrickingraham — COMMENTED** [review]
So some of that was pretty bad. My apologies.
We do have different expectations I think regarding how uhh "poetic" the document should be. 
I seem to be more on the "high level document" side whereas you seem to be more on the "technote" side. The answer is probably somewhere in between...
This push is a pretty significant re-write.... you might just want to re-read from scratch.

File: index.rst [hunk 20]
```diff
@@ -1,57 +1,343 @@
...
+
+To account for the large number of use-cases from all levels of users, the control system must simultaneously provide a high degree of flexibility and a user-friendly framework.
+The top-level functionality specifications that define much of the development of the observatory control system are:
+
+  - Regular operations routines should be easily accessible;
+  - Must include a highly customizable scripting capability must be available and able to run user-written code;
+  - An interactive mode, equivalent to a user sandbox space, must be available to allow development of routines on the fly as well as debugging;
+  - Capability to incorporate user-developed features;
+
+In the standard operational model used to perform the survey, the observatory will be largely driven by the observatory Scheduler.
```
Comment (tribeiro) [inline index.rst]: Was there supposed to be a comma between "model" and "used"?

Consider dropping the "observatory" before "Scheduler" as there's already an "observatory" earlier in the phrase, e.g.;

>In the standard operational model, used to perform the survey, the observatory will be largely driven by the Scheduler.

File: index.rst [hunk 21]
```diff
@@ -1,57 +1,343 @@
...
+The Scheduler designates a target and a specially formatted script containing a series of operations to be performed (such as slewing the telescope and taking an image).
+The execution of the script handled by the control system while the user is able to observe (or interrupt) the progress via a graphical interface.
```
Comment (tribeiro) [inline index.rst]: I think there is a missing **is** before handled. Also, consider "monitor" instead of "observe".

> The execution of the script **is** handled by the control system while the user is able to monitor (or interrupt) the progress via a graphical interface.

File: index.rst [hunk 22]
```diff
@@ -1,57 +1,343 @@
...
+To account for the large number of use-cases from all levels of users, the control system must simultaneously provide a high degree of flexibility and a user-friendly framework.
+The top-level functionality specifications that define much of the development of the observatory control system are:
+
+  - Regular operations routines should be easily accessible;
+  - Must include a highly customizable scripting capability must be available and able to run user-written code;
+  - An interactive mode, equivalent to a user sandbox space, must be available to allow development of routines on the fly as well as debugging;
+  - Capability to incorporate user-developed features;
+
+In the standard operational model used to perform the survey, the observatory will be largely driven by the observatory Scheduler.
+The Scheduler designates a target and a specially formatted script containing a series of operations to be performed (such as slewing the telescope and taking an image).
```
Comment (tribeiro) [inline index.rst:31]: Maybe add a comma after "formatted script"? 

> The Scheduler designates a target and a specially formatted script, containing a series of operations to be performed (such as slewing the telescope and taking an image).

File: index.rst [hunk 23]
```diff
@@ -1,57 +1,343 @@
...
+The execution of the script handled by the control system while the user is able to observe (or interrupt) the progress via a graphical interface.
+Operators will be much more involved during the start-up and shutdown phases, as well as during afternoon calibrations.
+In the majority of cases, these routine operations are expected to be executed by the operator from a high-level interface of the observatory control software (as discussed in following sections).
+
+During integration, commissioning, and engineering time, there will be situations where non-standard, customized operations will need to be executed.
+These operations will often be exercised by specialists to a specific subsystem (e.g. M1M3) and require a knowledge base that differs from that of an observing specialist.
```
Comment (tribeiro) [inline index.rst]: I really liked this phrase. One suggestion, would it make sense to replace "specialists to a specific subsystem"  by "subsystem specialist"? It will match really well with the later use of "observing specialist", e.g.:

> These operations will often be exercised by subsystem specialists (e.g. M1M3) and require a knowledge base that differs from that of an observing specialist.

File: index.rst [hunk 24]
```diff
@@ -1,57 +1,343 @@
...
+The breadth of the observatory control system allows users to perform both standard routines and customized tasks via three main interactive components:
+
+    - :ref:`LSST Observing Visualization Environment (LOVE)<LOVE>`
+    - :ref:`ScriptQueue <Script-Queue>`
+    - :ref:`Jupyter notebooks <Jupyter-notebooks>`
+
+This document begins with a high-level overview of the three components and explores how users can take advantage of their features to accomplish different types of tasks.
```
Comment (tribeiro) [inline index.rst]: "overview of the three components" -> "overview of these three components" ?

File: index.rst [hunk 25]
```diff
@@ -1,57 +1,343 @@
...
+Then, we define three main actors and outline their roles and assumed levels of knowledge regarding when performing specific tasks.
```
Comment (tribeiro) [inline index.rst]: "fun" fact, I learned that the use of many "and" in a phrase is called polysyndeton.

File: index.rst [hunk 26]
```diff
@@ -1,57 +1,343 @@
...
+These actors and roles are used to assist in defining the usability expectations of the components and control system in general as well as the underlying assumptions of the knowledge required for each role.
+In reality, the people's roles will be less discrete and there will be significant overlap between these roles.
+The identified actors are:
+
+    1. Observing Specialists
+    2. Commissioning Scientists
+    3. Software Developers
+
+Finally, we present concrete examples of interactions that the actors will perform with the OCS, linking to procedures on how to do them using the multiple components.
```
Comment (tribeiro) [inline index.rst]: I think this is the first (and only) time "OCS"  is used, whereas "observatory control system" is used broadly throughout the document . Do you want to replace OCS by "observatory control system" or introduce the acronym and use it throughout? I think I'd rather do the first.

File: index.rst [hunk 27]
```diff
@@ -1,57 +1,343 @@
...
+These examples explore how users best take advantage of the component's features.
+They also include cases of how *not* to use a component due limitations or inefficiencies in various areas.
```
Comment (tribeiro) [inline index.rst]: I think "not" should be in **bold**  rather than in *italic*. 

Also, add "to" between "due limitations"?

```
They also include cases of how **not** to use a component due to limitations or inefficiencies in various areas.
```

**Review by tribeiro — COMMENTED** [review]
Thank you for the updates! the first section definitely reads a lot better now. Added some additional comments to section 1.

File: index.rst [hunk 28]
```diff
@@ -1,57 +1,343 @@
...
+
+
+.. _Script-Queue:
+
+Script Queue
+============
+
+The `ScriptQueue`_ is the main hub for the execution of regular operations and requires a minimum level of interaction by the user.
+As mentioned in the previous section, it is also used as a back-end for user-commanded actions performed using the `LOVE`_ interface.
+For example, when a user performs a telescope offset operation using LOVE, the actual action perform by LOVE system will the queueing of a :ref:`SAL Script <SAL_Scripts>` that then gets executed by the `ScriptQueue`_.
```
Comment (tribeiro) [inline index.rst]: I think we need to clarify something here. LOVE can also send commands directly to CSCs that does not pass through the ScriptQueue. For instance, LOVE can acknowledge alarms from the Watcher directly. Also, the current "TCS" views will use the `TCS` class directly to execute tasks like offsets, stop and emergency stop.

File: index.rst [hunk 29]
```diff
@@ -1,57 +1,343 @@
...
+The `ScriptQueue`_ is flexible in how it handles and executes scripts but generally it runs sequentially.
+As soon as one script finishes, the next one begins until no scripts remain to be run.
+
 ..
-  Technote content.
+    This next bit feel repetitive..
 
-  See https://developer.lsst.io/restructuredtext/style.html
-  for a guide to reStructuredText writing.
+This architecture (or methodology) means the number functions that get executed by the scriptQueue will be very large and their complexity varies significantly.
```
Comment (tribeiro) [inline index.rst]: "the number functions" -> "the number of functions" ?

Not sure "functions" is the correct term here as it may be confused by functions in programing. Maybe "the number of operations"?

File: index.rst [hunk 30]
```diff
@@ -1,57 +1,343 @@
...
+For instance, when operators need to prepare the observatory for calibrations or for night operation, they will add a rather complex script to the ScriptQueue that will perform all the required tasks (e.g. position telescope and dome, perform required fine tuning in positioning, open mirror covers and so on).
```
Comment (tribeiro) [inline index.rst]: Should we use the term "observing specialists" introduced earlier for "operators"?

I wonder if they are the same "actor" or not.

File: index.rst [hunk 31]
```diff
@@ -1,57 +1,343 @@
...
 
-  Do not put the title, authors or other metadata in this document;
-  those are automatically added.
+The Scheduler for the survey essentially adds scripts to the scriptQueue after each visit until told by the observer to stop.
```
Comment (tribeiro) [inline index.rst:87]: "observer", "observing specialists" or "operator"?

File: index.rst [hunk 32]
```diff
@@ -1,57 +1,343 @@
...
+   :name: fig-scriptqueueui
+   :target: ../_images/ScriptQueueUI.png
+   :alt: LOVE ScriptQueue user interface
 
-  and
+   A screenshot of the LOVE interface to interact with the ScriptQueue.
 
-  Subsections
-  -----------
+In general, scripts require minimum interaction to be executed except, of course, for the occasional configuration.
```
Comment (tribeiro) [inline index.rst]: The first phrase of the section was:

> The ScriptQueue is the main hub for the execution of regular operations and requires a minimum level of interaction by the user. 

Maybe introduce the subject in a different way?

File: index.rst [hunk 33]
```diff
@@ -1,57 +1,343 @@
...
+A database of configurations will be available for users to execute the most common set of operations.
+Users also have the possibility to edit configurations from the user interface, which also provides on-the-fly schema validation.
 
-  and
+In general, using the ScriptQueue requires some familiarity with the observatory system and minimal set of software development skills.
```
Comment (tribeiro) [inline index.rst:106]: The previous paragraph also started with "In general" maybe start it differently? Maybe "Furthermore, ?"

File: index.rst [hunk 34]
```diff
@@ -1,57 +1,343 @@
...
-  _static/ directory. The reST syntax for adding the image is
+.. _SAL_Scripts:
 
-  .. figure:: /_static/filename.ext
-     :name: fig-label
+SAL Scripts
+-----------
 
-     Caption text.
+`SAL Scripts`_ are the files which contain the logic and coordination of events and CSCs that get executed by the `ScriptQueue`_.
```
Comment (tribeiro) [inline index.rst]: I think "files" is not exactly the term here. Maybe;

> SAL Scripts contain the logic and coordination of commands and events that get executed by the `ScriptQueue`_.

File: index.rst [hunk 35]
```diff
@@ -1,57 +1,343 @@
...
+It is not expected these will be modified during standard night-time operations.
+All `SAL Scripts`_ vary greatly in complexity of the operation being performed.
```
Comment (tribeiro) [inline index.rst]: Maybe start a new paragraph?

suggested edit:

> SAL Scripts vary considerably in complexity depending on the operation being performed.

File: index.rst [hunk 36]
```diff
@@ -1,57 +1,343 @@
...
+For example, a SAL script that performs a relatively fundamental task is `Enable MTCS <https://github.com/lsst-ts/ts_standardscripts/blob/develop/python/lsst/ts/standardscripts/maintel/enable_mtcs.py>`_ which brings all components of the TCS to the enabled state.
+The script is normally a launched using the default configuration which enables all components, however, the flexibility is present to only enable a subset if the observer chooses to do so.
```
Comment (tribeiro) [inline index.rst]: I think there's a rogue "a" here:

"The script is normally a launched" -> "The script is normally launched".

Maybe split the phrase into two?

> The script is normally a launched using the default configuration which enables all components. 
However, the flexibility is present to only enable a subset if the observer chooses to do so.

File: index.rst [hunk 37]
```diff
@@ -1,57 +1,343 @@
...
+A more complicated script, such as `Prepare for On-Sky <https://github.com/lsst-ts/ts_standardscripts/blob/develop/python/lsst/ts/standardscripts/auxtel/prepare_for_onsky.py>`_ performs a series of order-specific operations to bring the systems online, then open the dome and telescope safely.
+`SAL Scripts`_ obey strict formatting requirements and must contain specific information in order to be capable of execution.
```
Comment (tribeiro) [inline index.rst:126]: New paragraph? Also, I think this phrase could be removed altogether. I don't think it is adding any relevant information.

File: index.rst [hunk 38]
```diff
@@ -1,57 +1,343 @@
...
+Although normally executed via `LOVE`_, it is also possible to execute these from a Jupyter Notebook or from the command line when required.
```
Comment (tribeiro) [inline index.rst]: If you agree to remove the phrase above and start a new paragraph maybe rephrase this one a bit, e.g.;

> Although normally executed via `LOVE`_, it is also possible to execute SAL Scripts from a Jupyter Notebook or directly from the command line.

**Review by tribeiro — COMMENTED** [review]
Some comments on section 3

File: index.rst [hunk 39]
```diff
@@ -1,57 +1,343 @@
...
+.. _Jupyter-notebooks:
+
+Jupyter notebooks
+=================
+..
+    Do we need to mention Nublado in the text?
+
+The notebook server available at the summit control network is built on top of the `DM science platform`_, augmented with `Telescope and Site observatory control package`_.
+Notebooks allow users to combine observatory control activities with data analysis in a highly interactive web-based interface.
+This includes analysis of data queried the EFD.
```
Comment (tribeiro) [inline index.rst]: "... queried from the EFD"?

**Review by tribeiro — COMMENTED** [review]
Comment in section 4

File: index.rst [hunk 40]
```diff
@@ -1,57 +1,343 @@
...
+        - Includes operator control-software skills plus the following:
+        - Extensive use of the notebook interface, including the writing of code and launching of scripts
+
+            - Comfortable in Python, competent with git and GitHub
+
+        - Ability to diagnose both system and component level behavioural issues
+        - Not required to identify the issue in the source code
+        - Able to create and load new config files
+        - Writes and executes custom external scripts from both notebooks, ScriptQueue and LOVE
+        - Not expected to write production level scripts (see tstn-010 for definition)
```
Comment (tribeiro) [inline index.rst]: May want to link tstn-010.

File: index.rst (hunk 22 above)
Comment (patrickingraham) [reply index.rst:31]: I don't think there should be a comma there...

File: index.rst (hunk 25 above)
Comment (patrickingraham) [reply index.rst]: So we generally referred to them as run-on sentences... which are bad. I fixed it.

File: index.rst (hunk 26 above)
Comment (patrickingraham) [reply index.rst]: removed use of OCS... which might come back and bite us when this document gets really long... but we'll see

File: index.rst (hunk 28 above)
Comment (patrickingraham) [reply index.rst]: I knew that was true for the E-stop but not the others.
So what is a good example of a user-interaction that does load a script as opposed to using the TCS. Taking an image?

Comment (patrickingraham) [reply index.rst]: I changed the example to taking an image, and added a note on line 66 in the LOVE section to discuss the difference between when LOVE interacts with the TCS and when it interacts with the scriptQueue.

File: index.rst (hunk 29 above)
Comment (patrickingraham) [reply index.rst]: replaced with "number of different operations"

File: index.rst (hunk 31 above)
Comment (patrickingraham) [reply index.rst:87]: i used "observer" because it's super generic and can be any actor...

File: index.rst (hunk 32 above)
Comment (patrickingraham) [reply index.rst]: rephrased

File: index.rst (hunk 33 above)
Comment (patrickingraham) [reply index.rst:106]: rephrased the above line so this "In general" can stay.

File: index.rst (hunk 35 above)
Comment (patrickingraham) [reply index.rst]: Don't need a new paragraph... it's only the 3rd sentence... but I edited the phrase.

File: index.rst (hunk 37 above)
Comment (patrickingraham) [reply index.rst:126]: I think it's important as most people see a script as just a series of commands in a blank file... like a bash script.
I did make it a new paragraph

File: index.rst (hunk 38 above)
Comment (patrickingraham) [reply index.rst]: modified the phrase a bit, but didn't remove the top one (see response above)

**Comment by patrickingraham** [comment]
I think i responded to all the open conversations. Feel free to go through and resolved the ones that have been addressed.

File: index.rst [hunk 41]
```diff
@@ -1,57 +1,344 @@
...
+    - **Software Developer**
+
+        - Diagnoses behavioural issues at the code-level of CSCs or higher-level classes
+        - Writes and reviews production level scripts
+
+            - Expert in Python, git and other applicable languages
+
+        - Modifies and builds components, tags for release where appropriate
+        - Familiar with deployment strategies and restarting components (ArgoCD)
+        - Ability to probe into individually deployed containers (Rancher)
```
Comment (tribeiro) [inline index.rst]: Although the majority of our CSCs are deployed with containers on k8s we do have some special snowflakes out there. Maybe just state "components" in general?

> - Ability to probe into individually deployed components, in the kubernetes cluster and/or bare metal machines

File: metadata.yaml [hunk 42]
```diff
@@ -18,6 +18,7 @@ doc_title: "Concept of Control System Operations"
 # Author names, ordered as a list. Each author name should be formatted as 'First Last'
 authors:
   - "Tiago Ribeiro"
+  - "Patrick Ingraham"
```
Comment (tribeiro) [inline metadata.yaml:20]: Do you want to add your name first? after all you did most of the work here so far.

**Review by tribeiro — COMMENTED** [review]
Added comments on the remaining sections.

File: metadata.yaml (hunk 42 above)
Comment (patrickingraham) [reply metadata.yaml:20]: Sure... make me take the blame.... well played.
